oeps/best-practices/oep-0066-bp-authorization.rst

```diff
@@ -0,0 +1,395 @@
+   * - Created
+     - 2023-08-21
+   * - `Review Period`
+     - 2023-09-12 - 
+   * - Replaces
+     - :doc:`oep-0009-bp-permissions`
+   * - References
+     - 
+       * :doc:`/architectural-decisions/oep-0004-arch-oauth-scopes`
+       * :doc:`oep-0042-bp-authentication`
+     
+Summary
+*******
+
+There are a variety of manners in which authorization is handled within the Open edX ecosystem. 
+The goal of OEP-66 is to outline the currently in use systems/protocols and provide best practices 
+that should be used with all the systems/protocols.
+
+
+Defined Terms
+*************
+
+Authorization (Authz)
+---------------------
+
+Authorization is the granting of permission of a certain user to perform specific operations in an application. A user can also delegate an application to be authorized to perform operations on their behalf without being logged in or authenticated, which is the basis of OAuth.
+
+.. note::
+  The definition of authorization found here is the same as that found in 
+  :doc:`OEP 42 <oep-0042-bp-authentication>`.
+  Credit for this definition belongs to the authors of OEP-42; Robert Raposa, Nimisha Asthagiri, and Julia Eskew.
+
+Authentication (Authn)
+----------------------
+
+Authentication is the verification of the identity of a user, which typically initiates at a “login” application point. Authentication is required whenever we need to identify a client/caller/user/etc.
+
+.. note::
+
+  Authentication is out of scope of this OEP. 
+  The definition is included here to clarify the difference between it and Authorization.
+  The definition comes from :doc:`OEP 42 <oep-0042-bp-authentication>`.
+  Credit for this definition belongs to the authors of OEP-42; Robert Raposa, Nimisha Asthagiri, and Julia Eskew.
+
+RBAC
+--------------
+Role Based Access Control. A system in which roles are assigned to a user 
+in order to grant that user permission to perform specific operations.
+
+There are multiple RBAC implementations in use within Open edX, 
+including, but not limited to, edx-rbac and student_courseaccessrole.
+The implementations will be described in detail below.
+
+Explicit Role
+--------------
+A role that is specifically assigned to a user with 
+the intent of providing the user permission to perform specific operations.
+
+Implicit Role
+--------------
+A "role" that is understood to belong to a user based 
+on data that is not role assignment data. 
+
+Implicit roles grant users permissions, but are not specifically assigned 
+to a user.
+
+Best Practices
+***********************
+
+Permission Checks
+-----------------------
+
+Most authorization checks in Python code should use the standard
+`Django authorization API`_, including the optional support for object-level
+permissions which isn't implemented in the default backends.  Some examples
+can help give context for the details:
+
+.. _Django authorization API: https://docs.djangoproject.com/en/1.10/topics/auth/default/#permissions-and-authorization
```

Comment:
This doc is for Django 1.10 while edx-platform master uses Django 3.2. Should we use those newer docs?

Reply:
I agree this change makes sense. I'll update the code.